### ISSUE TITLE
Hyperion support (homemade ambilight)

### ISSUE THREAD

**Issue by digitalLumberjack** [digitalLumberjack/recalbox-os#36]

**Comment by Raulval82** [comment]
Yes it will be nice! I had it working under retropie and it's awsome!

**Comment by Raulval82** [comment]
Some more ideas for this:

I have two scripts for handling hyperion:

hyperionoff.sh that turns off the service, and it's called from autostart.sh
hyperionswitch.sh that allows you turning ON\OFF from a button (I have it working with tv remote controller, but here I think is better a button in emulationstation).

**Comment by armybean** [comment]
Any progress on this?

**Comment by Murlock7** [comment]
I just try Recalbox and it works great but I really need hyperion too ! if you want some help for testing i m ready !

**Comment by mxjoe** [comment]
+1

I really like Hyperion and am sure it should be integrated into recalbox. I have a Hyperion system myself and could assist in testing.

**Comment by mchtt** [comment]
i'm almost finished building my ambilight for osmc, i'd like to see it working on recalbox to :+1:

**Comment by Twitch0815** [comment]
Tried deploying installation script on Recalbox 4.0.0 beta.  


Runing ./install_hyperion.sh failed.

TP request sent, awaiting response… 200 OK

Length: 8037430 (7.7M) [application/octet-stream]

Saving to: ‘STDOUT’

 

–                     0%[                    ]       0  –.-KB/s    in 0.06s

 

 

Cannot write to ‘-‘ (Success).

ln: /usr/bin/hyperiond: Read-only file system

ln: /usr/bin/hyperion-remote: Read-only file system

ln: /usr/bin/hyperion-v4l2: Read-only file system

ln: /etc/hyperion.config.json: Read-only file system

 

You can replicate this without Hyperion hardware.

If you can create a build, with the following packages,

libqtcore4 libqtgui4 libqt4-network libusb-1.0-0 ca-certificates   and RW access to /usr/bin and /etc

I could attempt again for testing purposes, but this will likely need to be completely installed, and configured on the dev side.  Since this will recal a configuration file we would need r/w access to the configuration file,  or something standard in place such as. 

http://sdrv.ms/1c0uKK0

**Comment by substring** [comment]
you can't use the script as-is for several reasons : /opt doesn't exist on recalbox, and most of / is read-only

but you can still download yourself the binary and give it a go

**Comment by mxjoe** [comment]
If I chnage temporarily from read-only file system to read/write filesystem script could work. Do you agree?

**Comment by mxjoe** [comment]
Switched from read-only to read/write via `mount /flash -o remount,rw` BUT the script throws errors because there is no apt-get command available in Recalbox's buildroot system.

Is there a way to install apt-get?

I got my Hyperion/Ambilight solution running with OpenELEC and could assist in porting Hyperion to Recalbox.

**Comment by substring** [comment]
The script from.hyperion can't work for several reasons. Opt-get is one, the other one isnthat the system folders structure is not the same as raspbian.

I'll have a look at it sooner or later, although i don't even have any led system myself

**Comment by digitalLumberjack** [comment]
What do we do with this one ? Nobody in the dev have hyperion, and this issue is old now :)

**Comment by Raulval82** [comment]
Dont let it die please! If I can help with anything, I will.

**Comment by substring** [comment]
On my todo list ... Shall we ask for donations ?

**Comment by MathijsG** [comment]
Any news on this? @substring @Raulval82

**Comment by MikaXII** [comment]
@MathijsG Currently no, because nobody (team member or contributor) have the stuff to test and implement on recalbox.
